packages/react-magma-dom/src/components/Stepper/Step.tsx

```diff
@@ -199,68 +212,87 @@ const StyledSecondaryLabel = typedStyled.span<{
-        <StyledStepIndicator
-          hasError={hasError}
-          isInverse={isInverse}
-          stepStatus={stepStatus}
-          theme={theme}
-        >
-          {stepStatus === StepStatus.complete && !hasError && (
-            <CheckIcon aria-hidden="true" />
-          )}
-          {hasError && <CrossIcon aria-hidden="true" />}
-        </StyledStepIndicator>
+  return (
+    <StyledStep {...rest} ref={ref} data-testid={testId}>
+      <StyledStepIndicator
+        hasError={hasError}
+        isInverse={isInverse}
+        stepStatus={stepStatus}
+        theme={theme}
+      >
+        {stepStatus === StepStatus.complete && !hasError && (
+          <CheckIcon aria-hidden="true" />
+        )}
+        {hasError && <CrossIcon aria-hidden="true" />}
+      </StyledStepIndicator>
 
-        <StyledStepTextWrapper>
-          {!areLabelsHidden ? (
-            <>
-              {label && (
-                <StyledLabel
-                  label={label}
-                  isInverse={isInverse}
-                  data-testid={testId && `${testId}-label`}
-                  theme={theme}
-                >
-                  {label}
-                </StyledLabel>
-              )}
-              {secondaryLabel && (
-                <StyledSecondaryLabel
-                  secondaryLabel={secondaryLabel}
-                  isInverse={isInverse}
-                  data-testid={testId && `${testId}-secondaryLabel`}
-                  theme={theme}
-                >
-                  {secondaryLabel}
-                </StyledSecondaryLabel>
-              )}
-            </>
-          ) : (
-            <HiddenLabelText>
-              {label && secondaryLabel
-                ? `${label} ${secondaryLabel}`
-                : label || secondaryLabel}
+      <StyledStepTextWrapper>
+        {!areLabelsHidden ? (
+          <>
+            {layout === StepperLayout.showLabels && (
+              <HiddenLabelText>{`${stepLabel} ${index + 1}, `}</HiddenLabelText>
+            )}
+            {label && (
+              <StyledLabel
+                label={label}
+                isInverse={isInverse}
+                data-testid={testId && `${testId}-label`}
+                theme={theme}
+              >
+                {label}
+              </StyledLabel>
+            )}
+            {secondaryLabel && (
+              <StyledSecondaryLabel
+                secondaryLabel={secondaryLabel}
+                isInverse={isInverse}
+                data-testid={testId && `${testId}-secondaryLabel`}
+                theme={theme}
+              >
+                {secondaryLabel}
+              </StyledSecondaryLabel>
+            )}
+            {layout === StepperLayout.showLabels && (
+              <HiddenLabelText>
+                {stepStatus === StepStatus.complete
+                  ? `, ${stepLabel} ${stepStatus}`
+                  : ''}
+              </HiddenLabelText>
+            )}
```

Comment:
I'm not an avid VO user, but having this after the labels feels more clunky because there's more to navigate through. I wonder if we should consider having the status be part of the hidden label above (line 250). I guess I defer to what you both think.